### ISSUE TITLE
BUG: Provider Selection Logic

### ISSUE THREAD

**Issue by mjones-oddball** [department-of-veterans-affairs/notification-api#944]
## Issue
**This is blocking Phase 1 of the VEText Migration**

During work on [928](https://github.com/department-of-veterans-affairs/notification-api/issues/928) Cris identified an issue where Userflows were failing because sometimes the Pinpoint provider was chosen and sometimes the Twilio provider was chosen. If you're sending SMS you can't use just any provider as this is based on what provider owns the sender phone number.

In order to prevent issues in Production, we disabled the Twilio provider temporarily on 11/9.

## Endpoint
`/v2/notifications/sms`

## Goal
Given services cannot use just any provider for SMS:
1) Use the template's `provider_id` first
2) The service's `provider_id` if the template's `provider_id` is null
3) Ensure any other provider determinations are not made

Any other paths should be unreachable. We can work on stripping out the other paths at a later time. It is understand this will cause unit tests to fail. Comment out the ones that are failing because the paths can no longer be reached. Document all commented out tests and terminated paths so we can create a new ticket to clean it all up.

The tech debt ticket generated from this will be marked as high-priority next sprint so it's not left in the code base.

## Checklist
- [x] SMS uses the Template's `provider_id` first to determine the Provider
- [x] SMS uses the Service's `sms_provider_id` if the Template's `provider_id` is null
- [x] Other paths to acquire a Provider are not accessible
- [x] Update data in all envs to use Amazon as the `sms_provider_id` if it is currently null and the service has `sms` in their `permissions`
- [x] Create Tech Debt ticket to address what won't be done with this ticket
  - [ ] Document changes to pathing that causes tests to fail
  - [x] Document unreachable paths
~- [ ] Amazon Pinpoint should be our default SMS provider if a provider ID is not explicitly set on the service or template~
- [ ] Re-enable Twilio in production after the issue is resolved

## Additional Information
[Code block](https://github.com/department-of-veterans-affairs/notification-api/blob/master/app/provider_details/provider_service.py#L45) where we currently test for the template id having a `provider_id`:
![image.png](https://images.zenhubusercontent.com/620be841701e64d34dd34065/517b71cd-797c-49ad-bc19-b3cff72121a4)
Note: `_get_template_or_service_provider_id()` currently stores the logic to do items 1 and 2 in the Goal section.

Evan defined the current provider workflow here: https://github.com/department-of-veterans-affairs/notification-api/issues/942 
[PDF Copy](https://app.zenhub.com/files/257953872/7d526b3d-bec5-4e41-8ea1-1986a7b14f8a/download)

**Comment by mjones-oddball** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/VA-Notify-620d21369d810a00146ed9c8/issues/department-of-veterans-affairs/notification-api/944?planning-poker) with Zenhub @cris-oddball @EvanParish @ianperera @jakehova @k-macmillan @kalbfled @trevor2718 @ldraney

**Comment by kalbfled** [comment]
I managed to get a different [user flows failure](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/3484572938/jobs/5829681716).  I guess that's progress. :/

**Comment by EvanParish** [comment]
I've picked this up in Dave's absence. 
For code changes, I mainly cleaned up the logging sections that have been touched. 

I didn't manage to make it to the `Update data in all envs to use Amazon as the sms_provider_id ...` task. That will probably require running a query on the database for each environment to update the `sms_provider_id` on each service that has sms permissions. 

I think the `Document unreachable paths` task could be considered handled since I added TODOs to all the functions that should be removed. This work should be handled in the follow-up ticket #962.
